src/core/interaction-creator.ts

```diff
@@ -0,0 +1,206 @@
+import * as ts from 'typescript';
+import * as tsMorph from 'ts-morph';
+import {mapJsDocsIntoInteraction} from './js-docs-into-interaction';
+import {getBasicRepresentationOfType, getReturnTypeOfFunction} from './typescript-types';
+import {changeObjectRepresentationIntoExample, changeObjectRepresentationIntoMatchingRules} from './pact-generating';
+import {isEmptyObject} from '../utils/object-type';
+import qs from 'qs';
+import {Provider} from './read-pacts-config';
+import {getDefaultResponseStatusForInteraction} from './default-response-status';
+import {PACT_ANNOTATIONS} from '../consts/pact-annotations';
+
+export interface Interaction {
+    description?: string;
+    request: {
+        method?: string;
+        path?: string;
+        body?: unknown;
+        query?: string;
+        matchingRules?: object;
+        headers?: Record<string, string>;
+    };
+    response: {
+        status?: number;
+        body?: unknown;
+        matchingRules?: object;
+        headers?: Record<string, string>;
+    };
+}
+
+export class InteractionCreator {
+    constructor(private readonly sourceFile: tsMorph.Node, private readonly provider: Provider) {}
+
+    public static getAllInteractionsInFile(sourceFile: tsMorph.Node, provider: Provider) {
```

Comment:
This function looks like it is not necessary, it just gives and additional step during function calls flow.

Reply:
This is a main/start method for that class.
``public`` and ``static`` because we want use from this class without necessity of creating instance of class.

Reply:
But with every call of this function you are creating a new instance of InteractionCreator, so 
`interactions.push(...InteractionCreator.getAllInteractionsInFile(sourceFile, provider));`
is exactly the same as
`interactions.push(...new InteractionCreator(sourceFile, provider).findAllInteractions());`
but with additional step. It doesn't make anything clearer or more descriptive.